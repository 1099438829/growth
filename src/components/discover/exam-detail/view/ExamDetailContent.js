import React, { Component, PropTypes } from 'react';
import { View, Text, PanResponder } from 'react-native';
import Indicator from './ExamDetailIndicator';
import AppStyle from '../../../../theme/styles';

let startX;
let endX;
let pressPosition;
let leftslide;
let diff;

class ExamDetailContent extends Component {
  static componentName = 'ExamDetailContent';

  static propTypes = {
    quizs: PropTypes.arrayOf(
      PropTypes.string,
    ),
    onIndexChangeListener: PropTypes.func,
    test: PropTypes.bool,
  };

  static defaultProps = {
    quizs: [],
    onIndexChangeListener: () => {},
    test: false,
  };

  constructor(props) {
    super(props);
    this.state = {
      randomIndex: this.props.test
        ? 0
        : Math.floor(Math.random() * (this.props.quizs.length - 9)),
      text: this.props.quizs[
        this.props.test
          ? 0
          : Math.floor(Math.random() * (this.props.quizs.length - 9))],
      index: 0,
    };
  }

  componentWillMount() {
    this.panResponder = PanResponder.create({
      onStartShouldSetPanResponder: () => true,
      onStartShouldSetPanResponderCapture: () => true,
      onMoveShouldSetPanResponder: () => true,
      onMoveShouldSetPanResponderCapture: () => true,
      onPanResponderMove: () => true,
      onPanResponderStart: (e, gestureState) => {
        startX = gestureState.moveX;
        pressPosition = gestureState.x0;
        return true;
      },
      onPanResponderEnd: (e, gestureState) => {
        endX = gestureState.moveX;
        this.change();
        return true;
      },
    });
  }

  change() {
    const movePx = 0;
    const minCount = 0;
    const maxCount = 8;
    const minSlideValue = 30;
    if (endX === movePx) return;
    leftslide = (pressPosition - endX) > movePx;
    diff = endX - startX;
    if (!leftslide) {
      diff = -diff;
    }
    if (diff > minSlideValue && this.state.index < maxCount) {
      this.state.index += 1;
    } else if (diff < -minSlideValue && this.state.index > minCount) {
      this.state.index -= 1;
    }
    this.setState({
      text: this.props.quizs[this.state.randomIndex + this.state.index],
    });
    this.props.onIndexChangeListener(this.state.index);
  }

  render() {
    return (
      <View
        {...this.panResponder.panHandlers}
        style={{ flex: 1, backgroundColor: 'rgba(0,0,0,0)' }}
      >
        <View
          style={AppStyle.examDetailContentStyle}
        >
          <Text>{this.state.text}</Text>
        </View>
        <Indicator index={this.state.index} />
      </View>
    );
  }
}
export default ExamDetailContent;
