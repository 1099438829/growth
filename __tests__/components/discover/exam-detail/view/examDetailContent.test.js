import 'react-native';
import React, {} from 'react';
import renderer from 'react-test-renderer';
import ExamDetailContent from '../../../../../src/components/discover/exam-detail/view/ExamDetailContent';
import QUIZS from '../../../../../assets/data/QUIZS';

it('renders correctly', () => {
  const examDetailContent = renderer.create(
    <ExamDetailContent quizs={QUIZS.fe} test />,
  );
  const examDetailContentJson = examDetailContent.toJSON();
  expect(examDetailContentJson).toMatchSnapshot();
});
