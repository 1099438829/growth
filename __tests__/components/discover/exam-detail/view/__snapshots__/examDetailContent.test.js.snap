// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 50,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      原来公司工作流程是怎么样的，如何与其他人协作的？如何夸部门合作的？
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 8,
      }
    }
  >
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 15,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 15,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 15,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 15,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 15,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 15,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 15,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 15,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 15,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
